### ISSUE TITLE
p2p-circuit was not supported

### ISSUE THREAD

**Issue by godcong** [multiformats/go-multiaddr#109]
go-ipfs-http-client@v0.0.4 ->
swarm.go -> 
[out.addr, err = multiaddr.NewMultiaddr(conn.Addr) ](https://github.com/ipfs/go-ipfs-http-client/blob/187d8d16e5f7c53a1fb2039bf719499661b12926/swarm.go#L99)

if conn.Addr is a p2p-circuit address like
```
/ipfs/QmdGQoGuK3pao6bRDqGSDvux5SFHa4kC2XNFfHFcvcbydY/p2p-circuit/ipfs/QmcCnisUqm75ghwMSb8k1dJXvn9d6usK7pd9NzWqDZn8XG
```

this api will return an err
```
panic: failed to parse multiaddr "/ipfs/QmdGQoGuK3pao6bRDqGSDvux5SFHa4kC2XNFfHFcvcbydY/p2p-circuit": unknown protocol p2p-circuit
```

**Comment by Stebalien** [comment]
See https://github.com/multiformats/go-multiaddr/issues/108. The issue is that we need to import github.com/libp2p/go-libp2p-circuit to get the address definition.

**Comment by Stebalien** [comment]
Closing as a duplicate.

**Comment by godcong** [comment]
so quickly, 👍

**Comment by Stebalien** [comment]
@godcong do you have time to fix #108 and move the protocol definitions (https://github.com/libp2p/go-libp2p-circuit/blob/master/transport.go#L13-L24)? It's not tricky, just something we need to do.
